apps/guide/src/content/03-popular-topics/01-frequently-asked-questions.mdx

```diff
@@ -0,0 +1,495 @@
+	want to learn more about intents, check out [this dedicated guide on intents](/popular-topics/intents.md)!
+</Alert>
+
+### How do I check which role was added/removed and for which member?
+
+<CH.Code>
+
+```js
+// Start by declaring a guildMemberUpdate listener
+// This code should be placed outside of any other listener callbacks to prevent listener nesting
+client.on(Events.GuildMemberUpdate, (oldMember, newMember) => {
+	// If the role(s) are present on the old member object but no longer on the new one (i.e role(s) were removed)
+	const removedRoles = oldMember.roles.cache.filter((role) => !newMember.roles.cache.has(role.id));
+
+	if (removedRoles.size > 0) {
+		console.log(`The roles ${removedRoles.map((r) => r.name)} were removed from ${oldMember.displayName}.`);
+	}
+
+	// If the role(s) are present on the new member object but are not on the old one (i.e role(s) were added)
+	const addedRoles = newMember.roles.cache.filter((role) => !oldMember.roles.cache.has(role.id));
+
+	if (addedRoles.size > 0) {
+		console.log(`The roles ${addedRoles.map((r) => r.name)} were added to ${oldMember.displayName}.`);
+	}
+});
+```
+
+</CH.Code>
+
+### How do I check the bot's ping?
+
+There are two common measurements for bot pings. The first, **websocket heartbeat**, is the average interval of a regularly sent signal indicating the healthy operation of the websocket connection the library receives events over:
+
+<CH.Code>
+
+```js
+await interaction.reply(`Websocket heartbeat: ${client.ws.ping}ms.`);
+```
+
+</CH.Code>
+
+<Alert title="Tip" type="info">
+	If you're using [sharding](/sharding/), a specific shard's heartbeat can be found on the WebSocketShard instance,
+	accessible at _`client.ws.shards.get(id).ping`_.
+</Alert>
+
+The second, **Roundtrip Latency**, describes the amount of time a full API roundtrip (from the creation of the command message to the creation of the response message) takes. You then edit the response to the respective value to avoid needing to send yet another message:
+
+<CH.Code>
+
+```js
+const sent = await interaction.reply({ content: 'Pinging...', fetchReply: true });
+await interaction.editReply(`Roundtrip latency: ${sent.createdTimestamp - interaction.createdTimestamp}ms`);
+```
+
+</CH.Code>
+
+### Why do some emojis behave weirdly?
+
+If you've tried using [the usual method of retrieving unicode emojis](./reactions.md#unicode-emojis), you may have noticed that some characters don't provide the expected results. Here's a short snippet that'll help with that issue. You can toss this into a file of its own and use it anywhere you need! Alternatively feel free to simply copy-paste the characters from below:
+
+<CH.Code>
+
+```js index.js
+import { emojiCharacters } from './emojiCharacters.js';
+
+console.log(emojiCharacters.a); // 🇦
+console.log(emojiCharacters[10]); // 🔟
+console.log(emojiCharacters['!']); // ❗
+```
+
+```js emojiCharacters.js
+// prettier-ignore
```

Comment:
This might be a better solution since it wont show up on the website itself
```suggestion

<!-- prettier-ignore -->
```js emojiCharacters.js
```

Reply:
Unfortunately not... preview throws errors when this happens. I did a bit of experimenting around and saw that placing `{/* prettier-ignore */}` worked instead! See 780d11ab9bdd177ce184e80d807aee5c8e45a069.